### PR TITLE
Fix copying of teaser selection in WebspaceCopyCommand

### LINKED ISSUE #4539: WebspaceCopyCommand does not copy teaser selection properties
| Q | A
| --- | ---
| Bug? | yes
| New Feature? | no
| Sulu Version | 1.6
| Browser Version | irrelevant

#### Actual Behavior

Executing the webspace copy command does not copy 'teaser selection' content types.

#### Expected Behavior

It should copy it, just like the 'block' types.

#### Steps to Reproduce

Create two webspaces, then create a page type with a teaser_selection content type. Execute the webspace copy command, and observe that the teaser selection content type is not copied.

#### Possible Solutions

This case is not handled here:

https://github.com/sulu/sulu/blob/b8bdb0185b1e2f1e81fd560cab1b6ee012bfce36/src/Sulu/Bundle/ContentBundle/Command/WebspaceCopyCommand.php#L417

### PR DESCRIPTION
**Pull request by debreczeniandras**
| Q | A
| --- | ---
| Bug fix? | yes
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | fixes #4539
| License | MIT

#### What's in this PR?

Process teaser_selection content type.

#### To Do

- [x] Write tests upon request

### CODE REVIEW

File: src/Sulu/Bundle/ContentBundle/Command/WebspaceCopyCommand.php [hunk 1]
```diff
@@ -535,6 +543,40 @@ protected function updateSmartContentStructure(
...
+        $localeSource,
+        $localeDestination
+    ) {
+        if (!array_key_exists($property->getName(),
+                $structureArray) || !$structureArray[$property->getName()]['items']) {
+            return;
+        }
+
+        foreach ($structureArray[$property->getName()]['items'] as $key => $teaserItem) {
+            $targetDocumentDestination = $this->getTargetDocumentDestination(
```
Comment (alexander-schranz) [inline src/Sulu/Bundle/ContentBundle/Command/WebspaceCopyCommand.php]: is here not a check missing todo this only for `type` page?

Comment (alexander-schranz) [reply src/Sulu/Bundle/ContentBundle/Command/WebspaceCopyCommand.php]: Not sure where the type is saved but it shoud something like:
foreach (...) {
if ('page' !== $teaserItem['type']) {
     continue;
}

Comment (debreczeniandras) [reply src/Sulu/Bundle/ContentBundle/Command/WebspaceCopyCommand.php]: I couldn't yet find it, but why would you want to limit which type of teasers should be processed? 

The `getTargetDocumentDestination()` method should be able to resolve any (even custom) Teaser types based on the uuid. 
No?

Comment (alexander-schranz) [reply src/Sulu/Bundle/ContentBundle/Command/WebspaceCopyCommand.php]: It should skip other types like `article` or any other custom entity.

My example is false the type for pages is `content` so it should be something like:

```
foreach (...) {
     if ('content' !== $teaserItem['type']) {
         continue;
     }

     // ...
}
```

Comment (debreczeniandras) [reply src/Sulu/Bundle/ContentBundle/Command/WebspaceCopyCommand.php]: @alexander-schranz it's now breaking from the loop in case it's not 'content'.

Comment (alexander-schranz) [reply src/Sulu/Bundle/ContentBundle/Command/WebspaceCopyCommand.php]: looks correct for me. should work for pages (content) still correctly or?

Comment (debreczeniandras) [reply src/Sulu/Bundle/ContentBundle/Command/WebspaceCopyCommand.php]: yes, it does work. already tested it.

File: src/Sulu/Bundle/ContentBundle/Command/WebspaceCopyCommand.php [hunk 2]
```diff
@@ -535,6 +543,44 @@ protected function updateSmartContentStructure(
...
+     * @param string $localeSource
+     * @param string $localeDestination
+     */
+    protected function updateTeaserSelection(
+        array &$structureArray,
+        PropertyMetadata $property,
+        $localeSource,
+        $localeDestination
+    ) {
+        if (!array_key_exists($property->getName(),
```
Comment (alexander-schranz) [inline src/Sulu/Bundle/ContentBundle/Command/WebspaceCopyCommand.php]: @debreczeniandras sorry for a change request again. Did test it on my current webspace and we need to change also this lines to:

```
        if (!isset($structureArray[$property->getName()]['items'])) {
            return;
        }
```

as in some cases the property exist but could be null and isset would be enough for this case and error proof.

**Comment by alexander-schranz** [comment]
@debreczeniandras and can you update your branch to the current master branch and add a line to the `CHANGELOG.md`: 

```
    * BUGFIX      #4540  [ContentBundle]          Fix copying of teaser selection in WebspaceCopyCommand
```

File: src/Sulu/Bundle/ContentBundle/Command/WebspaceCopyCommand.php (hunk 2 above)
Comment (debreczeniandras) [reply src/Sulu/Bundle/ContentBundle/Command/WebspaceCopyCommand.php]: @alexander-schranz no problem, it's now changed.

Comment (alexander-schranz) [reply src/Sulu/Bundle/ContentBundle/Command/WebspaceCopyCommand.php]: @debreczeniandras  can you add the CHANGELOG.md entry also .

**Comment by danrot** [comment]
Shouldn't we also add a test for this? :thinking:

**Comment by alexander-schranz** [comment]
Thank you @debreczeniandras 
It is released in version 1.6.27

**Pull request merged**